File: BOJ(Java 공부용)/1551.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), k = Integer.parseInt(st.nextToken());
        String[] input = br.readLine().split(",");
        int[] arr = new int[n];
        for (int i = 0; i < input.length; i++) arr[i] = Integer.parseInt(input[i]);
        for (int i = 0; i < k; i++) {
            int[] temp = new int[n];
            int idx = 0;
            for (int j = 0; j < arr.length - 1; j++) {
                temp[idx++] = arr[j + 1] - arr[j];
            }
            arr = temp;
        }
        for (int i = 0; i < arr.length - k; i++) {
            if (i == arr.length - k - 1) {
                bw.write(arr[i] + "\n");
            } else bw.write(arr[i] + ",");
        }
        bw.flush();
        bw.close();
    }
}

File: BOJ(Python 공부용)/1551.py
import sys
from math import sqrt

input = sys.stdin

if __name__ == '__main__':
    n, k = map(int, input.readline().split())
    v = [int(x) for x in input.readline().split(",")]

    for i in range(k):
        temp = []
        for j in range(len(v)):
            if j == len(v) - 1: break
            temp.append(v[j + 1] - v[j])

        v = temp

    for i in range(len(v)):
        if i == len(v) - 1:
            print(v[i], end="\n")
        else:
            print(v[i], end=",")
